run-test.js

```diff
@@ -0,0 +1,72 @@
+// const exec =  require('child_process').exec;
+const { spawn, exec } = require('node:child_process');
+const {echo} = require('shelljs');
+const minimist = require("minimist");
+const chalk = ("chalk");
+
+const log = console.log;
+// const error = chalk.bold.red;
+// const debug = chalk.bold.green;
+// const info = chalk.rgb(215, 135, 95);
+
+echo("Running Script")
+exec("jest -u --colors --changedSince feature/patial-test") //this works fine
+
+
+function getPRDetails() {
+    let prDetails;
+exec("gh pr view", (err, stdout, stderr) => {
+  if (stderr || err) {
+    console.error("Failed to fetch PR details. Install Github CLI and login in order to get base branch / No open PR for this branch");
+    return;
+  }
+  const response = stdout.toString();
+  prDetails = response.match(/into (\d+) from (\d+)/i);
+});
+    return prDetails;
+  }
+getPRDetails();
+  async function fetchBaseBranch(baseBranch) {
```

Comment:
I needed this function on JenkinsCI. you most probably don't need this since you already checkout the branch in tests.yml